### PR TITLE
Switch to kotlin

### PR DESCRIPTION
**Pull request by hwbllmnn**
@terrestris/devs Please review.

### CODE REVIEW

**Review by dnlkoch — APPROVED** [review]
Awesome!

File: src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        @DescribeParameter(name = "filter", description = "An optional CQL filter to apply.", min = 0) filter: String?,
+        @DescribeParameter(name = "viewParams", description = "Optional view params.", min = 0) viewParams: String?
+    ): RawData {
+        var conn: Connection? = null
+        var stmt: PreparedStatement? = null
+        var rs: ResultSet? = null
+        return try {
+            val factory = JsonNodeFactory(false)
+            val root = factory.arrayNode()
+            val tableName = layerName.split(":".toRegex()).toTypedArray()[1]
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt]:
```suggestion
            val tableName = layerName.split(":".toRegex())[1]
```
don't know if this is better

File: src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        filter: String?
+    ): String {
+        var sql = virtualTable!!.sql
+        if (viewParams != null) {
+            val params = viewParams.split(";".toRegex()).toTypedArray()
+            for (param in params) {
+                val parts = param.split(":".toRegex()).toTypedArray()
+                sql = sql.replace("%" + parts[0] + "%", parts[1])
+            }
+        }
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt:144]:
```
        if (viewParams != null) {
            val params = viewParams.split(";".toRegex())
            for (param in params) {
                val parts = param.split(":".toRegex())
                sql = sql.replace("%" + parts[0] + "%", parts[1])
            }
        }
```

**Review by simonseyock — COMMENTED** [review]
Cool :) some nitpicks

File: src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt (hunk 1 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt]: Right, this is not necessary, I'm not sure why the conversion generated it like that

File: src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt (hunk 2 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt:144]: Changed in 66b19d5

**Pull request merged**